test/unit/lists-component.spec.ts

```diff
@@ -563,3 +569,160 @@ describe('when deleting a list', () => {
+        created_at: new Date()
+      }
+
+      dbQueryMock.mockResolvedValueOnce({ rowCount: 1, rows: [dbList] })
+      result = await listsComponent.getList(listId, { userAddress, considerDefaultList: true })
+    })
+
+    it('should have made the query to get the list checking if the list belongs to the default user without taking into account the permissions', async () => {
+      expect(dbQueryMock).toHaveBeenCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining('SELECT *')
+        })
+      )
+
+      expect(dbQueryMock).toHaveBeenCalledWith(
+        expect.objectContaining({
+          text: expect.stringContaining('FROM favorites.lists WHERE id = $1 AND (user_address = $2 OR user_address = $3)'),
+          values: expect.arrayContaining([listId, userAddress, DEFAULT_LIST_USER_ADDRESS])
+        })
+      )
+    })
+
+    it('should resolve', () => {
+      return expect(result).toEqual(dbList)
+    })
+  })
+
+  describe('and both the default list and the permissions should be considered', () => {
+    let dbList: DBList
+    let result: DBList
+
+    describe.each([Permission.EDIT, Permission.VIEW])('and the required permission is %s', permission => {
+      beforeEach(async () => {
+        dbList = {
+          id: 'aListId',
+          name: 'aListName',
+          description: null,
+          user_address: 'aUserAddress',
+          created_at: new Date(),
+          permission
+        }
+
+        dbQueryMock.mockResolvedValueOnce({ rowCount: 1, rows: [dbList] })
+        result = await listsComponent.getList(listId, { userAddress, considerDefaultList: true, requiredPermissions: [permission] })
+      })
+
+      it('should have made the query to get the list matching those conditions', async () => {
+        expect(dbQueryMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            text: expect.stringContaining('SELECT *, favorites.acl.permission as permission')
+          })
+        )
+
+        expect(dbQueryMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            text: expect.stringContaining('FROM favorites.lists WHERE id = $1 AND (user_address = $2 OR user_address = $3)'),
+            values: expect.arrayContaining([listId, userAddress, DEFAULT_LIST_USER_ADDRESS])
+          })
+        )
+
+        expect(dbQueryMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            text: expect.stringContaining('LEFT JOIN favorites.acl ON favorites.lists.id = favorites.acl.list_id')
+          })
+        )
+
+        expect(dbQueryMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            text: expect.stringContaining(
+              'WHERE (favorites.acl.grantee = $4 OR favorites.acl.grantee = $5) AND favorites.acl.permission = ANY($6::text[])'
+            ),
+            values: expect.arrayContaining([userAddress, '*', [permission]])
+          })
+        )
+      })
+
+      it('should resolve', () => {
```

Comment:
What if we add here the `with the list` phrase to make the assertion clearer?
```suggestion
      it('should resolve with the list', () => {
```